plugins/crd/crd_queue.go

```diff
@@ -49,39 +53,53 @@ func networkserviceWork(plugin *Plugin) {
 		// We define a function here to process a queue item, so that we can
 		// use 'defer' to make sure the message is marked as Done on the queue
 		func(key string) {
-			var obj interface{}
-			var err error
-
 			defer queueNS.Done(key)
 
 			// Attempt to split the 'key' into namespace and object name
 			namespace, name, err := cache.SplitMetaNamespaceKey(strKey)
 
 			if err != nil {
+				// This is a soft-error, retry up to QUEUE_RETRY_COUNT times
 				plugin.Log.Errorf("Error splitting meta namespace key into parts: %s", err.Error())
-				runtime.HandleError(fmt.Errorf("Error splitting meta namespace key into parts: %s", err.Error()))
-				// This is a soft-error, we merely want to forget it and mark it done on the queue
-				queueNS.Forget(key)
+				if err != nil {
+					if queueNS.NumRequeues(key) < QUEUE_RETRY_COUNT {
```

Comment:
Why would this condition be an error if it returns less than the retry count? The documentation reads as follows:

```go
	// NumRequeues returns back how many failures the item has had
	NumRequeues(item interface{}) int
```

Reply:
The idea is we're flagging the fact it's an error, and then re-queueing the item. I'll change the error message in this particular case.